rllib/connectors/connector_v2.py

```diff
@@ -149,8 +127,232 @@ def __call__(
+            # number of agents and M is the number of modules and N >= M.
+            for eps in episodes[:-1]:
+                eps.agent_to_module_map = {"agent_0": "module_0", "agent_1": "module_1"}
+            # Only the last episode maps the other way around.
+            episodes[-1].agent_to_module_map = {
+                "agent_0": "module_1", "agent_1": "module_0"
+            }
+
+            # Now, our batch would look like this:
+            batch = {
+                "obs": {
+                    "agent_0": [0, 1, 2],  # <- could also be ndarray of shape (3,)
+                    "agent_1": [0, 1],  # <- could also be ndarray of shape (2,)
+                },
+            }
+            # Note that the data under the different agent IDs might have different
+            # batch dimensions (or a different length in the case of lists) b/c not
+            # every episode might have the same agent ID distribution (some agents
+            # might not even be present in some episodes).
+
+            flipped_batch = ConnectorV2.switch_batch_from_agent_ids_to_module_ids(
+                batch, episodes
+            )
+            check(
+                flipped_batch,
+                {
+                    "obs": {
+                        "module_0": [0, 1],
+                        "module_1": [0, 1, 2],
+                    },
+                },
+            )
+
+        Args:
+            batch: The batch to switch from an agent ID based mapping to a module ID
+                based mapping.
+            #episodes: The list of MultiAgentEpisodes to use to extract agent-to-module
+            #    mapping information. Note that the
+        """
+
+    @staticmethod
+    def switch_batch_from_column_to_module_ids(batch):
+        """Switches the first two levels of a `col -> ModuleID -> data` type batch.
+
+        Assuming that the top level consists of column names as keys and the second
+        level (under these columns) consists of ModuleID keys, the resulting batch
+        will have these two reversed and thus map ModuleIDs to dicts mapping column
+        names to data items.
+
+        .. testcode::
+
+            batch = {
+                "obs": {"module_0": [1, 2, 3]},
+                "actions": {"module_0": [4, 5, 6], "module_1": [7]},
+            }
+            switched_batch = ConnectorV2.switch_batch_from_column_to_module_ids(batch)
+
+            check(
+                switched_batch,
+                {
+                    "module_0": {"obs": [1, 2, 3], "actions": [4, 5, 6]},
+                    "module_1": {"actions": [7]},
+                },
+            )
+
+        Args:
+            batch: The batch to switch from being column name based (then ModuleIDs)
+                to being ModuleID based (then column names).
         """
+        module_data = {}
+        for column, column_data in batch.items():
+            for module_id, data in column_data.items():
+                module_data[module_id]
```

Comment:
This function does not return anything - it looks to me incomplete. Is this intended? 

Also, couldn't we just use a dict comprehension here? Like:
```python
return {
       module_id: {
               column: data
               for module_id, data in column_data.items()
        } for column, column_data in batch.items()
}
```